src/main/java/ua/yakaboo/pages/HeaderPage.java

```diff
@@ -1,4 +1,158 @@
+    private WebElement blogButton;
+
+    @FindBy(xpath = ".//div[@class='elementor-image']//a[contains(@href,'blog')]")
+    private WebElement blogPage;
+
+    public HeaderPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public void waitPromotionsButtonToBeClickable(WebDriver driver) {
+        Waits waits = new Waits(driver);
+        waits.waitElementToBeClickableShort(promotionsButton);
+    }
+
+    public void clickPromotionsButtonWithAction(WebDriver driver) {
+        Actions actions = new Actions(driver);
+        actions.click(promotionsButton).build().perform();
+        Waits waits = new Waits(driver);
+        waits.waitVisibilityOfElement(discountsPage, 3);
+    }
+
+    public void clickCancelButtonOnDialog(WebDriver driver) {
+        WebElement shadowRootElement = (WebElement) ((JavascriptExecutor) driver).executeScript("return arguments[0].shadowRoot", attachedWebElement);
+        shadowRootElement.findElement(shadowOkButton).click();
+    }
+
+    public void waitAboutUsButtonToBeClickable(WebDriver driver) {
+        Waits waits = new Waits(driver);
+        waits.waitElementToBeClickableShort(aboutUsButton);
+    }
+
+    public void clickAboutUsButton(WebDriver driver) {
+        aboutUsButton.click();
+        Waits waits = new Waits(driver);
+        waits.waitVisibilityOfElementShort(aboutUsPage);
+    }
+
+    public void waitDeliveryButtonToBeClickable(WebDriver driver) {
+        Waits waits = new Waits(driver);
+        waits.waitElementToBeClickableShort(deliveryButton);
+    }
+
+    public void clickDeliveryButton(WebDriver driver) {
+        deliveryButton.click();
+        Waits waits = new Waits(driver);
+        waits.waitVisibilityOfElementShort(deliveryPage);
+    }
+
+    public void waitPaymentButtonToBeClickable(WebDriver driver) {
+        Waits waits = new Waits(driver);
+        waits.waitElementToBeClickableShort(paymentButton);
+    }
+
+    public void clickPaymentButton(WebDriver driver) {
+        paymentButton.click();
+        Waits waits = new Waits(driver);
+        waits.waitVisibilityOfElementShort(paymentPage);
+    }
+
+    public void waitContactButtonToBeClickable(WebDriver driver) {
+        Waits waits = new Waits(driver);
+        waits.waitElementToBeClickableShort(contactButton);
+    }
+
+    public void clickContactButton(WebDriver driver) {
+        contactButton.click();
+        Waits waits = new Waits(driver);
+        waits.waitVisibilityOfElementShort(contactPage);
+    }
+
+    public void waitBlogButtonToBeClickable(WebDriver driver) {
+        Waits waits = new Waits(driver);
+        waits.waitElementToBeClickableShort(blogButton);
+    }
+
+    public void clickBlogButton() {
+        blogButton.click();
+    }
+
+    public boolean isStoreLogoDisplayed() {
+        return storeLogo.isDisplayed();
+    }
+
+    public boolean isDiscountsPageDisplayed() {
+        return discountsPage.isDisplayed();
```

Comment:
You can create isElementDisplayed() method in Waits class, and use it anywhere.

Reply:
Created